src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,154 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    public static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(T item) {
+            this.item = item;
+        }
+
+    }
+
+    public void insertAtTheBeginningOfTheList(T value) {
```

Comment:
lets make all helper methods private (remember private methods should follow after public)